lib/ansible/modules/cloud/vmware/vmware_vspan_session.py

```diff
@@ -0,0 +1,582 @@
+                    port = vim.dvs.VmwareDistributedVirtualSwitch.VspanPorts(uplinkPortName=str(self.destination_port))
+                    session.destinationPort = port
+            if self.session_type == 'remoteMirrorDest':
+                if self.source_port_received is not None:
+                    port = vim.dvs.VmwareDistributedVirtualSwitch.VspanPorts(vlans=[int(self.source_port_received)])
+                    session.sourcePortReceived = port
+                if self.destination_port is not None:
+                    port = vim.dvs.VmwareDistributedVirtualSwitch.VspanPorts(portKey=str(self.destination_port))
+                    session.destinationPort = port
+            if self.session_type == 'dvPortMirror':
+                if self.source_port_received is not None:
+                    port = vim.dvs.VmwareDistributedVirtualSwitch.VspanPorts(portKey=str(self.source_port_received))
+                    session.sourcePortReceived = port
+                if self.source_port_transmitted is not None:
+                    port = vim.dvs.VmwareDistributedVirtualSwitch.VspanPorts(portKey=str(self.source_port_transmitted))
+                    session.sourcePortTransmitted = port
+                if self.destination_port is not None:
+                    port = vim.dvs.VmwareDistributedVirtualSwitch.VspanPorts(portKey=str(self.destination_port))
+                    session.destinationPort = port
+        if self.description is not None:
+            session.description = self.description
+        if self.encapsulation_vlan_id is not None:
+            session.encapsulationVlanId = self.encapsulation_vlan_id
+        if self.strip_original_vlan is not None:
+            session.stripOriginalVlan = self.strip_original_vlan
+        if self.mirrored_packet_length is not None:
+            session.mirroredPacketLength = self.mirrored_packet_length
+        if self.normal_traffic_allowed is not None:
+            session.normalTrafficAllowed = self.normal_traffic_allowed
+        if self.sampling_rate is not None:
+            session.samplingRate = self.sampling_rate
+        config_version = self.dv_switch.config.configVersion
+        s_spec = vim.dvs.VmwareDistributedVirtualSwitch.VspanConfigSpec(vspanSession=session, operation="add")
+        c_spec = vim.dvs.VmwareDistributedVirtualSwitch.ConfigSpec(vspanConfigSpec=[s_spec], configVersion=config_version)
+        task = self.dv_switch.ReconfigureDvs_Task(c_spec)
+        try:
+            wait_for_task(task)
+        except Exception:
+            self.restore_original_state()
+            self.module.fail_json(msg=task.info.error.msg)
+
+    def add_vspan_session(self):
+        """Calls the necessary functions to create a VSpanSession"""
+        results = dict(changed=False, result="")
+        promiscous_ports = self.turn_off_promiscuous()
+        if not self.check_if_session_name_is_free():
+            self.module.fail_json(msg='There is another VSpan Session with this name.')
```

Comment:
Suggest to add the actual name into the message.
```suggestion
            self.module.fail_json(msg="There is another VSpan Session with the name: {0:s}.".format(self.name))
```